src/v2/client/startLoginFlow.js

```diff
@@ -37,49 +38,64 @@ export async function startLoginFlow(settings) {
-    if (configuredFlow && configuredFlow === CONFIGURED_FLOW.PROCEED) {
-      const meta = await getSavedTransactionMeta(settings);
-      if (!meta) {
+    let meta = await authClient.idx.getSavedTransactionMeta();
+    const idxOptions = {
+      version: idxApiVersion,
+      exchangeCodeForTokens: false // we handle this in interactionCodeFlow.js
+    };
+    if (!meta) {
+      // no saved transaction
+      // if the configured flow is set to `proceed`, the SIW should only continue an existing idx transaction
+      // if the SIW loads from a fresh state (there is no current transaction), throw an error
+      const flow = authClient.idx.getFlow();
+      if (flow && flow === CONFIGURED_FLOW.PROCEED) {
         throw new Errors.ConfiguredFlowError(
-          'Unable to proceed: saved transaction could not be loaded', configuredFlow
+          'Unable to proceed: saved transaction could not be loaded', flow
         );
       }
+      // start new transaction
+      return authClient.idx.start(idxOptions);
     }
 
-    return interact(settings);
+    // continue saved transaction
+    return authClient.idx.proceed(idxOptions);
   }
 
   // Use stateToken from session storage if exists
   // See more details at ./docs/use-session-token-prior-to-settings.png
   const stateHandleFromSession = sessionStorageHelper.getStateHandle();
   if (stateHandleFromSession) {
-    return introspect(settings, stateHandleFromSession)
-      .then((idxResp) => {
-        // 1. abandon the settings.stateHandle given session.stateHandle is still valid
-        settings.set('stateToken', stateHandleFromSession);
-        // 2. chain the idxResp to next handler
-        return idxResp;
-      })
-      .catch(() => {
-        // 1. remove session.stateHandle
-        sessionStorageHelper.removeStateHandle();
-        // 2. start the login again in order to introspect on settings.stateHandle
-        return startLoginFlow(settings);
+    try {
+      const idxResp = await authClient.idx.introspect({
+        stateHandle: stateHandleFromSession,
+        version: idxApiVersion
       });
+      const hasError = idxResp.context?.messages?.value.length > 0;
+      if (hasError) {
+        throw new Error('saved stateToken is invalid'); // will be caught in this function
+      }
+
+      // 1. abandon the settings.stateHandle given session.stateHandle is still valid
+      settings.set('stateToken', stateHandleFromSession);
+      // 2. chain the idxResp to next handler
+      return idxResp;
+    } catch {
+      // 1. remove session.stateHandle
+      sessionStorageHelper.removeStateHandle();
+      // 2. start the login again in order to introspect on settings.stateHandle
+      return startLoginFlow(settings);
+    }
   }
 
   // Use stateToken from options
   const stateHandle = settings.get('stateToken');
   if (stateHandle) {
-    return introspect(settings, stateHandle);
+    return authClient.idx.introspect({ stateHandle, version: idxApiVersion });
```

Comment:
is `version` mandatory to pass to `authClient.idx` ? otherwise we can omit it so that easy to deprecated `appState. apiVersion`? or even remove `appState.apiVersion` in this PR?

Reply:
Yeah, I think we can remove `apiVersion` in this PR. If we do need to change the version in the future this will happen internally in auth-js, I don't think we will be passing this on from a widget option.